src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,184 @@
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertFalse(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void ifPasswordLessThanSixCharacters_NotOk() {
+        user.setPassword(LESS_THAN_6_CHARACTERS);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertFalse(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void ifAgeLessThan18_NotOk() {
+        user.setAge(UNDERAGE);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertFalse(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void testNullLogin_NotOk() {
+        user.setLogin(NULL_STRING);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertFalse(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void testNullPassword_NotOk() {
+        user.setPassword(NULL_STRING);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertFalse(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void testNullAge_NotOk() {
+        user.setAge(NULL_INTEGER);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertFalse(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void testAgeZero_NotOk() {
+        user.setAge(ZERO_AGE);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertFalse(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void ifLoginExactlySixCharacters_Ok() {
+        user.setLogin(SIX_CHARACTER_STRING);
+        registrationService.register(user);
+        assertTrue(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void ifPasswordExactlySixCharacters_Ok() {
+        user.setPassword(SIX_CHARACTER_STRING);
+        registrationService.register(user);
+        assertTrue(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void ifAgeExactlyEighteen_Ok() {
+        user.setAge(AGE_EIGHTEEN);
+        registrationService.register(user);
+        assertTrue(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void ifLoginIsEmpty_NotOk() {
+        user.setLogin(EMPTY_STRING);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertFalse(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void ifPasswordIsEmpty_NotOk() {
+        user.setPassword(EMPTY_STRING);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertFalse(registrationService.isSuccesfullyRegistered(user));
```

Comment:
The test 'ifPasswordIsEmpty_NotOk' correctly checks for empty password. Ensure that the test name follows the convention `<methodUnderTest>_<state>_<expectedBehavior>`.